### DISCUSSION TITLE
what is the difference between `VSCodeNotifyRange` and `VSCodeNotifyRangePos`?

### DISCUSSION THREAD

**Discussion by MrOxMasTer** [vscode-neovim/vscode-neovim#2259] [Q&A]
It's all wonderful of course that you want to switch completely to the Lua API, but such is the problem that I don't understand what the point of these functions are, or rather how they differ

I have some code that I borrowed from another person because I am not very good with neovim / lua and I would like to know the difference between these functions, because I am not that `old` of this extension and was not from the bottom of the VimScript API, but this piece of `deprecated` information is not enough for me, and on the Lua API it is not clear at all how they differ:
![image](https://github.com/user-attachments/assets/7b80156f-b893-41fe-85ec-63c173f66354)

I would like to understand this API completely to fully understand how the function works.

![image](https://github.com/user-attachments/assets/38ab7bea-8f93-49f7-af6e-b7b8fcf6a73e)
Specifically, for example, what the last parameter in the `VSCodeNotifyRangePos` function is responsible for.

**Comment by justinmk** [comment]
> Specifically, for example, what the last parameter in the `VSCodeNotifyRangePos` function is responsible for

That is the `leaveSelection` param: https://github.com/vscode-neovim/vscode-neovim/blob/4df0f6e3a61285ed7e090ceb6d1253288a4b092f/runtime/vscode-neovim.vim#L121

With the Lua api which replaces that api, that parameter would be passed as `restore_selection` item in the `opts` options map: https://github.com/vscode-neovim/vscode-neovim/blob/4df0f6e3a61285ed7e090ceb6d1253288a4b092f/runtime/lua/vscode/api.lua#L98-L109

So, an example call using [the Lua api](https://github.com/vscode-neovim/vscode-neovim?tab=readme-ov-file#vscodeactionname-opts) (just guessing based on your screenshot; please post code instead of screenshots):

```lua
require('vscode').call('workbench.action.showCommands', {
  range = {
    start = { line = startPos[1], character = endPos[1] },
    end = { line = startPos[2], character = endPos[2]},
  },
  restore_selection = true,
})
```
